src/main/kotlin/dongyeon-lee/230307/P43238.kt

```diff
@@ -0,0 +1,22 @@
+package `dongyeon-lee`.`230307`
+
+/* 킹국갓사 */
+class Solution {
+    fun solution(n: Int, times: IntArray): Long {
+        var answer: Long = 0
+
+        var min: Long = 1
+        var max: Long = n.toLong() * times.maxOrNull()!!
+        while (min <= max) {
+            val mid = (min + max) / 2
+            val people = times.fold(0L) { acc, time -> acc + (mid / time) }
```

Comment:
이거랑 같은 코드같네요(?)
fold 쓸필요가 없을것같아요.

```suggestion
            val people = times.sumOf { mid / it }
```